helm/values.schema.json

```diff
@@ -0,0 +1,1856 @@
+                "template": "keybase",
+                "receiver_url": "https://bots.keybase.io/webhookbot/<Your-Keybase-Hook-Token>",
+                "fail_if_alert_sending_fails": true
+              }
+            ]
+          }
+        }
+      }
+    ],
+    "required": [
+      "deployment",
+      "service",
+      "validators",
+      "policy"
+    ],
+    "properties": {
+      "deployment": {
+        "$id": "#/properties/deployment",
+        "type": "object",
+        "title": "The deployment schema",
+        "description": "Configuration of the Connaisseur Kubernetes deployment.",
+        "examples": [
+          {
+            "replicasCount": 3,
+            "image": "securesystemsengineering/connaisseur:v2.2.0",
+            "imagePullPolicy": "IfNotPresent",
+            "failurePolicy": "Fail",
+            "resources": {
+              "limits": {
+                "cpu": "1000m",
+                "memory": "512Mi"
+              },
+              "requests": {
+                "cpu": "100m",
+                "memory": "128Mi"
+              }
+            },
+            "nodeSelector": {},
+            "tolerations": [],
+            "affinity": {
+              "podAntiAffinity": {
+                "preferredDuringSchedulingIgnoredDuringExecution": [
+                  {
+                    "podAffinityTerm": {
+                      "labelSelector": {
+                        "matchExpressions": [
+                          {
+                            "key": "app.kubernetes.io/instance",
+                            "operator": "In",
+                            "values": [
+                              "connaisseur"
+                            ]
+                          }
+                        ]
+                      },
+                      "topologyKey": "kubernetes.io/hostname"
+                    },
+                    "weight": 100
+                  }
+                ]
+              }
+            },
+            "securityContext": {
+              "allowPrivilegeEscalation": false,
+              "capabilities": {
+                "drop": [
+                  "ALL"
+                ]
+              },
+              "privileged": false,
+              "readOnlyRootFilesystem": true,
+              "runAsNonRoot": true,
+              "runAsUser": 10001,
+              "runAsGroup": 20001,
+              "seccompProfile": {
+                "type": "RuntimeDefault"
+              }
+            },
+            "podSecurityPolicy": {
+              "enabled": false,
+              "name": [
+                "connaisseur-psp"
+              ]
+            },
+            "envs": {}
+          }
+        ],
+        "required": [
+          "replicasCount",
+          "image",
+          "imagePullPolicy",
+          "resources",
+          "nodeSelector",
+          "tolerations",
+          "affinity",
+          "securityContext",
+          "podSecurityPolicy",
+          "envs"
```

Comment:
are those actually required? bc I believe at least some of them can be dropped and will have some type of defaults, isn't it?

Reply:
but I am not sure which policy was decided upon

Reply:
no, you're right - these are not actually required, but we don't set any kind of default (but k8s does) and we once talked about it and decided to require these values for that reason, but rethinking it, I changed my mind and we should not require them if leaving them out does not yield a non-functional installation, what do you think?

Reply:
I removed the non-required values here and made the defaults explicit where I found defaults - if you don't like it I can change it back :)